File: desafio-hotmart-back-end/src/main/java/br/com/hotmart/desafiohotmart/vo/MessageInfoVO.java
package br.com.hotmart.desafiohotmart.vo;

import br.com.hotmart.desafiohotmart.enumerations.ReturnTypeEnum;

/**
 * Classe responsável por conter as mensgens de retorno
 * de uma requisição.
 * 
 * @author Tiago
 *
 */
public class MessageInfoVO implements BaseVO {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = -3707788901966746251L;

	private String message;
	
	private ReturnTypeEnum returnType;
	
	/**
	 * Construtor default.
	 * 
	 */
	public MessageInfoVO() {
		super();
	}

	/**
	 * Construtor que recebe uma mensagem e um tipo de retorno.
	 * 
	 * @param message
	 * @param returnType
	 */
	public MessageInfoVO(String message, ReturnTypeEnum returnType) {

		this.message = message;
		this.returnType = returnType;
		
	}

	/**
	 * @return the message
	 */
	public String getMessage() {
		return message;
	}

	/**
	 * @param message the message to set
	 */
	public void setMessage(String message) {
		this.message = message;
	}

	/**
	 * @return the returnType
	 */
	public ReturnTypeEnum getReturnType() {
		return returnType;
	}

	/**
	 * @param returnType the returnType to set
	 */
	public void setReturnType(ReturnTypeEnum returnType) {
		this.returnType = returnType;
	}

}

File: desafio-hotmart-front-end/src/app/models/message-info-vo.ts
import { TipoRetornoEnum } from '../enum/tipo-retorno-enum';

export class MessageInfoVO {
  
    private _message : string;
  
    private _returnType : TipoRetornoEnum;
  

    public get message(): string {

        return this._message;

    }

    public set message(message: string) {

        this._message = message;

    }

    public get returnType(): TipoRetornoEnum {

        return this._returnType;

    }

    public set returnType(returnType: TipoRetornoEnum) {

        this._returnType = returnType;

    }  
  
}
